src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/MarshalAsMarshallingGeneratorFactory.cs

```diff
@@ -144,10 +144,10 @@ private static IMarshallingGenerator CreateCharMarshaller(TypePositionInfo info,
                 {
                     case CharEncoding.Utf16:
                         return s_utf16Char;
-                    case CharEncoding.Ansi:
-                        throw new MarshallingNotSupportedException(info, context) // [Compat] ANSI is not supported for char
+                    case CharEncoding.Utf8:
```

Comment:
Can you remind me why we don't care about the rest of the `CharEncoding` values?

Reply:
`Undefined`: means no marshalling info
`Ansi`: since we got rid of `CharSet`, there isn't a way to specify ANSI anymore for char
`Custom`: I realized that this plays weirdly with char. We probably want to use `StringMarshalling.Custom`/`StringMarshallingCustomType` for string-only (we'll currently use the custom marshaller for both string and char, which leads to confusion and badness) and say that it is unsupported for char.